thrust/examples/dot_products_with_zip.cu

```diff
@@ -12,16 +12,16 @@
 // into a single virtual Float3 array.
 
 // We'll use a 3-tuple to store our 3d vector type
-using Float3 = thrust::tuple<float, float, float>;
+using Float3 = cuda::std::tuple<float, float, float>;
 
 // This functor implements the dot product between 3d vectors
 struct DotProduct
 {
   __host__ __device__ float operator()(const Float3& a, const Float3& b) const
   {
-    return thrust::get<0>(a) * thrust::get<0>(b) + // x components
-           thrust::get<1>(a) * thrust::get<1>(b) + // y components
-           thrust::get<2>(a) * thrust::get<2>(b); // z components
+    return cuda::std::get<0>(a) * cuda::std::get<0>(b) + // x components
+           cuda::std::get<1>(a) * cuda::std::get<1>(b) + // y components
+           cuda::std::get<2>(a) * cuda::std::get<2>(b); // z components
```

Comment:
Suggestion: we could really use a structured binding here.

Reply:
I would rather not change any code with this PR